### ISSUE TITLE
Nested FlatList does not scroll

### ISSUE THREAD

**Issue by rzinnatullin** [facebook/react-native#30925]
## Description
I'm nesting a FlatList inside a FlatList, both vertically oriented. The nested FlatList doesn't scroll.
I noticed that if I replace the nested FlatList with a ScrollView, the nested ScrollView works as expected.

## React Native version:
System:
    OS: Windows 10 10.0.19041
    CPU: (12) x64 Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
    Memory: 16.57 GB / 31.85 GB
  Binaries:
    Node: 12.14.1 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.5 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 6.13.4 - C:\Program Files\nodejs\npm.CMD
    Watchman: Not Found
  SDKs:
    Android SDK:
      API Levels: 23, 24, 25, 26, 27, 28, 29, 30
      Build Tools: 28.0.3, 29.0.2, 30.0.2
      System Images: android-30 | Google APIs Intel x86 Atom, android-30 | Google Play Intel x86 Atom
      Android NDK: Not Found
    Windows SDK: Not Found
  IDEs:
    Android Studio: Version  4.0.0.0 AI-193.6911.18.40.6821437
    Visual Studio: 16.8.30717.126 (Visual Studio Enterprise 2019)
  Languages:
    Java: 1.8.0_262
    Python: 3.8.6
  npmPackages:
    @react-native-community/cli: Not Found
    react: 16.13.1 => 16.13.1
    react-native: 0.63.4 => 0.63.4
    react-native-windows: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

## Steps To Reproduce
1. Nest a FlatList inside a FlatList, both vertically oriented

## Expected Results
The nested FlatList should scroll (like ScrollView does)

## Snack, code example, screenshot, or link to a repository:
```
import React from 'react';
import {Text, FlatList, View} from 'react-native';

const data = [
  'one',
  'two',
  'three',
  'four',
  'five',
  'six',
  'seven',
  'eight',
  'nine',
  'ten',
];

const App: () => React$Node = () => {
  const renderLevelTwo = (title, subTitle) => {
    return (
      <Text
        style={{
          height: 30,
          backgroundColor: 'lime',
          margin: 5,
          paddingHorizontal: 10,
        }}>
        {title} - {subTitle}
      </Text>
    );
  };

  const renderLevelOne = (title) => {
    return (
      <View style={{backgroundColor: 'green', borderWidth: 2}}>
        <View style={{height: 70, zIndex: 10}}>
          <FlatList
            renderItem={(itemInfo) => renderLevelTwo(title, itemInfo.item)}
            keyExtractor={(item) => item}
            data={data}
            scrollEnabled={true}
            nestedScrollEnabled={true}
          />
        </View>
        <View style={{height: 30, backgroundColor: 'yellow', zIndex: 5}} />
      </View>
    );
  };

  return (
    <View style={{height: 300}}>
      <FlatList
        renderItem={(itemInfo) => renderLevelOne(itemInfo.item)}
        keyExtractor={(item) => item}
        data={data}
        scrollEnabled={true}
        nestedScrollEnabled={true}
      />
    </View>
  );
};

export default App;
```
I also noticed that the nested FlatList overflows the parent area:
![image](https://user-images.githubusercontent.com/11244866/107466804-82d94980-6b86-11eb-941f-4114183c628c.png)

https://snack.expo.io/@xirustam/nested-flatlist

**Comment by rzinnatullin** [comment]
There's the same issue https://github.com/facebook/react-native/issues/15375 which wasn't resolved, but closed.

**Comment by SandyOnReact** [comment]
can u try setting nestedScrollEnabled={true}

**Comment by tastafur** [comment]
I have the same problem when I nest two flatLists and in the one inside I try to scrollToIndex, it gives an error in the VirtualizedList on line 452 in this._scrollRef.scrollTo, the scroll functions do not have them for whatever the flatList is the one I want to scroll.

React-native version: 0.63.4

**Comment by pratikadh** [comment]
Try importing Flatlist from 'react-native gesture handler'.
`import { FlatList } from 'react-native-gesture-handler';`
Check this [question](https://stackoverflow.com/questions/69585468/flatlist-exceeds-its-container-with-its-content-how-to-make-flatlist-with-fixed/69923245#69923245) in stackoverflow.

**Comment by HenrisHub** [comment]
> Try importing Flatlist from 'react-native gesture handler'. `import { FlatList } from 'react-native-gesture-handler';` Check this [question](https://stackoverflow.com/questions/69585468/flatlist-exceeds-its-container-with-its-content-how-to-make-flatlist-with-fixed/69923245#69923245) in stackoverflow.

For some reason this works!
